src/core/lib/event_engine/posix_engine/posix_engine_listener.h

```diff
@@ -189,33 +204,71 @@ class PosixEngineListener
       : impl_(std::make_shared<PosixEngineListenerImpl>(
             std::move(on_accept), std::move(on_shutdown), config,
             std::move(memory_allocator_factory), poller, std::move(engine))) {}
-  ~PosixEngineListener() override { impl_->TriggerShutdown(); };
+  ~PosixEngineListener() override { ShutdownListeningFds(); };
   absl::StatusOr<int> Bind(
       const grpc_event_engine::experimental::EventEngine::ResolvedAddress& addr)
       override {
-    return impl_->Bind(addr);
+    return impl_->Bind(addr, nullptr);
+  }
+  absl::StatusOr<int> BindWithFd(
+      const EventEngine::ResolvedAddress& addr,
+      PosixListenerWithFdSupport::OnPosixBindNewFdCallback on_bind_new_fd)
+      override {
+    return impl_->Bind(addr, std::move(on_bind_new_fd));
+  }
+  absl::Status HandleExternalConnection(int listener_fd, int fd,
+                                        SliceBuffer* pending_data) override {
+    return impl_->HandleExternalConnection(listener_fd, fd, pending_data);
   }
   absl::Status Start() override { return impl_->Start(); }
 
+  void ShutdownListeningFds() override {
+    if (!shutdown_.exchange(true, std::memory_order_acq_rel)) {
+      impl_->TriggerShutdown();
+    }
+  }
+
  private:
   std::shared_ptr<PosixEngineListenerImpl> impl_;
+  // Set to true when the listener had been explicitly shutdown.
+  std::atomic<bool> shutdown_{false};
 };
 
 #else  // GRPC_POSIX_SOCKET_TCP
 
-class PosixEngineListener
-    : public grpc_event_engine::experimental::EventEngine::Listener {
+class PosixEngineListener : public PosixListenerWithFdSupport {
```

Comment:
Why do we need this supporting nothing?

Reply:
This is in-case the target is built on non-posix systems. An attempt to use the object should fail.

Reply:
Is compile-time error preferred or runtime error?

Reply:
It already would generate a compile time error. The windows build/checks will fail without this. I need to do some build cleanup once event engine is landed to make sure this doesn't get pulled in windows builds. We should be able to remove this at that point.

Reply:
By default all targets declared in src/core/BUILD and in BUILD will be built on all platforms by the kokoro checks.